lib/src/features/checkout/view/buy_button.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import '../../../constants/type_defs/type_defs.dart';
+import 'checkout_controller.dart';
+import '../../authentication/data/authentication_repository.dart';
+import '../model/item.dart';
+import '../model/order.dart';
+
+import '../../../widgets/dialog/alert_dialog.dart';
+
+class BuyButton extends ConsumerWidget {
+  const BuyButton({super.key, required this.courseId, required this.price});
+
+  final CourseId courseId;
+  final double price;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isSignedIn = ref.watch(isSignedInProvider);
+    return SizedBox(
+      width: double.infinity,
+      child: FilledButton(
+        onPressed: isSignedIn
```

Comment:
em viết ra 1 function signInRequired(VoidCallback `func`) => `func` là hàm sẽ cần verify sign in trước khi gọi nha

Reply:
Em sửa bên #40 rồi nha